tests/ticket/search.rb

```diff
@@ -108,6 +285,399 @@
+
+        result = request('/ticket/add-tag', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            tagId: 2,
+            ticketNumber: @ticketNumber3
+        })
+        (result['status']).should.equal('success')
+
+        result = request('/ticket/add-tag', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            tagId: 3,
+            ticketNumber: @ticketNumber3
+        })
+        (result['status']).should.equal('success')
+
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            tags: "[2]"
+        })
+        (result['status']).should.equal('success')
+
+        tickets = result['data']['tickets']
+
+        (tickets.length()).should.equal(2)
+
+        for ticket in tickets
+            if ticket['tags'].length() === 2
+                (ticket['tags']).should.equal([tagNameOfId2, tagNameOfId3])
+            else
+                (ticket['tags']).should.equal([tagNameOfId2])
+            end
+        end
+
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            tags: "[3]"
+        })
+        (result['status']).should.equal('success')
+
+        tickets = result['data']['tickets']
+        (tickets.length()).should.equal(2)
+
+        for ticket in tickets
+            if ticket['tags'].length() === 2
+                (ticket['tags']).should.equal([tagNameOfId2, tagNameOfId3])
+            else
+                (ticket['tags']).should.equal([tagNameOfId3])
+            end
+        end
+
+
+        result = request('/ticket/search', {
+            csrf_userid: $csrf_userid,
+            csrf_token: $csrf_token,
+            page: 1,
+            tags: "[2, 3]"
+        })
+        (result['status']).should.equal('success')
+
+        for ticket in tickets
+            if ticket['tags'].length() === 2
+                (ticket['tags']).should.equal([tagNameOfId2, tagNameOfId3])
+            elsif ticket['tags'] === [tagNameOfId3]
+                (ticket['tags']).should.equal([tagNameOfId3])
+            elsif ticket['tags'] === [tagNameOfId2]
+                (ticket['tags']).should.equal([tagNameOfId2])
+            else
+                1.shoul.equal(0)
+            end
+        end
+    end
+
+    it 'should success if the departments are valid' do
+        for page in $pages
+            result = request('/ticket/search', {
+                csrf_userid: $csrf_userid,
+                csrf_token: $csrf_token,
+                page: page,
+                departments: "[1]"
+            })
+            (result['status']).should.equal('success')
+
+            tickets = result['data']['tickets']
+
+            for ticket in tickets
+                (ticket['department']['id']).should.equal('1')
+            end
+        end
```

Comment:
I would add more tests here, at least one with a department different than 1 (if we have one), and probably check things with private departments.